game/src/game/attack/mod.rs

```diff
@@ -161,37 +173,69 @@ pub fn attack_damage(
-                if health.current == 0 {
-                    commands.entity(entity).insert(Dead);
+
+        let mut collided: Vec<usize> = Vec::new();
+        for entity in colliding_entities.iter() {
+            let mut should_apply_damage = true;
+            for (i, attack) in attack.damaged.iter_mut().enumerate() {
+                if attack.entity == *entity {
+                    collided.push(i);
+                    if attack.contact == true {
+                        should_apply_damage = false;
+                    }
                 }
-                if let Some(pushback) = maybe_pushback {
-                    commands.entity(entity).insert(Knockback::new(
-                        pushback.direction,
-                        pushback.strength,
-                        16,
-                    ));
+            }
+            if should_apply_damage {
+                if let Ok((entity, mut health, collider, gent, is_defending)) =
+                    damageable_query.get_mut(*entity)
+                {
+                    let damage_dealt = if is_defending {
+                        attack.damage / 4
+                    } else {
+                        attack.damage
+                    };
+                    health.current = health.current.saturating_sub(damage_dealt);
+                    collided.push(attack.damaged.len());
+                    attack.damaged.push(DamageInfo {
+                        entity,
+                        tick: time.tick(),
+                        amount: damage_dealt,
+                        contact: true,
+                    });
+                    if let Ok((anim_entity, mut anim_player)) = gfx_query.get_mut(gent.e_gfx) {
+                        // is there any way to check if a slot is set?
+                        anim_player.set_slot("Damage", true);
+                        commands.entity(anim_entity).insert(DamageFlash {
+                            current_ticks: 0,
+                            max_ticks: 8,
+                        });
+                    }
+                    if health.current == 0 {
+                        commands.entity(entity).insert(Dead);
+                    }
+                    if let Some(pushback) = maybe_pushback {
+                        commands.entity(entity).insert(Knockback::new(
+                            pushback.direction,
+                            pushback.strength,
+                            16,
+                        ));
+                    }
                 }
             }
         }
+
+        // Finds all stored prev attacks, sees if that target entity was attacked
+        // and if not, marks its contact info as false.
+
+        collided.reverse();
+        for (i, attack) in attack.damaged.iter_mut().enumerate() {
+            if *collided.last().unwrap_or(&usize::MAX) == i {
+                collided.pop();
+            } else {
+                attack.contact = false;
+            }
+        }
+
```

Comment:
The biggest change in this pr; attacks now can trigger twice on the same enemy if the attack stops colliding and starts colliding again.
This code here is part of it that resets the tracking of entities no longer colliding.